libgalois/include/katana/PropertyIndex.h

```diff
@@ -0,0 +1,88 @@
+#ifndef KATANA_LIBGALOIS_KATANA_PROPERTYINDEX_H_
+#define KATANA_LIBGALOIS_KATANA_PROPERTYINDEX_H_
+
+#include <bitset>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <arrow/api.h>
+#include <arrow/chunked_array.h>
+#include <arrow/type_traits.h>
+#include <boost/iterator/iterator_categories.hpp>
+#include <boost/iterator/iterator_facade.hpp>
+
+#include "katana/config.h"
+#include "katana/PropertyGraph.h"
+#include "katana/Result.h"
+
+namespace katana {
+
+// PropertyIndex provides an interface similar to an ordered container
+// over a single property.
+class KATANA_EXPORT PropertyIndex {
+public:
+  // PropertyIndex::iterator returns a sequence of node or edge ids.
+  using node_iterator = std::set<GraphTopology::Node>::iterator;
+  using edge_iterator = std::set<GraphTopology::Edge>::iterator;
+
+  // Create an index from a column of pg.
+  static katana::Result<std::unique_ptr<PropertyIndex>>
+    Make(PropertyGraph* pg, std::string column);
+
+  std::string column() { return column_; }
+
+  virtual node_iterator node_begin() = 0;
+  virtual node_iterator node_end() = 0;
+
+  virtual edge_iterator edge_begin() = 0;
+  virtual edge_iterator edge_end() = 0;
+
+private:
+  std::string column_;
```

Comment:
Is this the name/alias of the index or the name of the column being indexed

Reply:
Name of the column being indexed.